### ISSUE TITLE
Wrong error location in traceback inside an assert

### ISSUE THREAD

**Issue by solanav** [python/cpython#116034]
# Bug report

### Bug description:

The location in the traceback for this failed assertion:

```python
test = 3
assert test == 1 and test == 2, "Bug found?"
```

Seems to be wrong. It should show `test == 1` as the source of the error:

```pytb
Traceback (most recent call last):
  File "/home/.../test.py", line 3, in <module>
    assert test == 1 and test == 2, "Bug found?"
                         ^^^^^^^^^
AssertionError: Bug found?
```

### CPython versions tested on:

3.11

### Operating systems tested on:

Linux

### Linked PRs
* gh-116052
* gh-116054

**Comment by Eclips4** [comment]
Hello!
Thanks for the report!
Traceback for current 3.11 branch:
```python
Traceback (most recent call last):
  File "/Users/admin/Projects/cpython/example.py", line 2, in <module>
    assert test == 1 and test == 2, "Bug found?"
                         ^^^^^^^^^
AssertionError: Bug found?
```
Traceback for current 3.12 branch:
```python
Traceback (most recent call last):
  File "/Users/admin/Projects/cpython/example.py", line 2, in <module>
    assert test == 1 and test == 2, "Bug found?"
AssertionError: Bug found?
```
Traceback for current main branch:
```python
Traceback (most recent call last):
  File "/Users/admin/Projects/cpython/example.py", line 2, in <module>
    assert test == 1 and test == 2, "Bug found?"
           ^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Bug found?
```

cc @pablogsal

**Comment by pablogsal** [comment]
Error locations are set by the compiler so this is likely a compiler bug. CC @iritkatriel

**Comment by iritkatriel** [comment]
in 3.13 it does this: 

```
>>> assert test == 1 and test == 2, "Bug found?"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
    assert test == 1 and test == 2, "Bug found?"
           ^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Bug found?
```

**Comment by pablogsal** [comment]
Shouldn't this be just one or the other?

**Comment by iritkatriel** [comment]
> Shouldn't this be just one or the other?

Since the line numbers are calculated statically, that would require emitting a different RAISE instruction for each clause. Currently we don't.

**Comment by pablogsal** [comment]
Then it seems there isn't much we can do for 3.13, no? For 3.12 it seems that we don't emit line info

**Comment by iritkatriel** [comment]
I fixed 3.11 and 3.12 to behave like 3.13. I will close this issue as the problem it reports it resolved.

I'm not sure narrowing the ranges is worth more code. If it ever comes up as an actual problem, we can revisit.

Thank you @solanav .
